frontend/app/src/components/asset-manager/CustomAssetFormDialog.vue

```diff
@@ -0,0 +1,81 @@
+<script setup lang="ts">
+import { type Ref } from 'vue';
+import { type CustomAsset } from '@/types/asset';
+import CustomAssetForm from '@/components/asset-manager/CustomAssetForm.vue';
+import { checkBeforeSubmission } from '@/utils/validation';
+
+const props = withDefaults(
+  defineProps<{
+    value: boolean;
+    title: string;
+    subtitle?: string;
+    types?: string[];
+    editableItem?: CustomAsset | null;
+  }>(),
+  {
+    subtitle: '',
+    types: () => [],
+    editableItem: null
+  }
+);
+
+const emit = defineEmits<{
+  (e: 'input', open: boolean): void;
+  (e: 'saved', assetId: string): void;
+}>();
+
+const { value, editableItem } = toRefs(props);
+
+const valid: Ref<boolean> = ref(true);
+const saving: Ref<boolean> = ref(false);
+const form: Ref<InstanceType<typeof CustomAssetForm> | null> = ref(null);
+
+const clearDialog = () => {
+  emit('input', false);
+};
+
+const confirmSave = () => {
+  checkBeforeSubmission(save, get(form)?.v$, valid);
+};
+
+const save = async () => {
+  if (!isDefined(form)) {
+    return;
+  }
+  set(saving, true);
+  const assetId = await get(form).save();
+  if (assetId) {
+    clearDialog();
+    emit('saved', assetId);
+  }
+  set(saving, false);
+};
```

Comment:
Is there any specific reason why we can't move the `save` function back to the form itself?

Reply:
You mean this whole function `save` right?

Because the button to `save` is in the `<big-dialog>`, not in the `form`, 
and we modify the `saving` state here (to disable the button)

Reply:
do you think we can follow a similar approach like here https://github.com/rotki/rotki/blob/201662111d866c1a27ef0dcbd38c26a070a06188/frontend/app/src/composables/accounts/dialog.ts#L11

We can extract the commonly shared logic to a composable, make it shared so that they both share the same instance, and remove the ref in the parent. Otherwise, the changes are in the right direction.

Reply:
Ah okay, I think we can do it better that way, will do it.